### ISSUE TITLE
Button 8 question.

### ISSUE THREAD

**Issue by sushi2503** [fin-ger/logitech-m720-config#13]
Hello,

I'm about to receive an M720 and saw your script, thanks for that. But I don't understand what does the sentence "This script will map Button 8 on the thumb button by default" mean ?

Can you explain it ?

And in the owner manual (see below) the thumb button is 7 and 8 is "dual connectivity" : 

PRODUCT FEATURES
1 Easy-switch
2 Forward button
3 Back button
4 Toggle & connect button
5 Hyper-fast scroll-wheel
6 Battery LED
7 Thumb button
8 Dual Connectivity
9 Battery & Receiver storage

Thank you.

**Comment by fin-ger** [comment]
You listed product features, not mouse buttons.

The mouse buttons are enumerated by the driver. The button 8 is the thumb button. This button is not working under Linux. In fact, by default it is sending some Keyboard commands which screw up your desktop. My script can deactivate this behavior and map the mouse button to one of the other buttons on the mouse, e.g. button 5. This makes the thumb button behave like button 5. However, you then have 2 buttons on your mouse doing the same thing.

The mouse is originally designed to come along with the Logitech Options App. It is not available for Linux and afaik Logitech has no plans to port it to Linux. Therefore the official user manual will be of no help for you. Neither can you expect to get any help from Logitech regarding Linux support.

My script only makes their product usable and the thumb button not completely useless under Linux. One might expect a huge company like Logitech being able to do this by themselves, but apparently they have no interest in doing so.

TL;DR
None of the advertised programmability features of this mouse are supported by Logitech under Linux. If you were excited to use them, I fear that most of them wont work. There are some third-party community projects with mixed support for this mouse which might achieve some of those features. But remeber those are community projects and not officially supported by Logitech. There is no guarantee that anything of this will work.

**Comment by sushi2503** [comment]
Thanks a lot for your explanations fin-ger !

Now I see it very clear and sorry for the confusion between the features mentioned and buttons numbers.

I already own a Logitech mouse (M705) with the Linux useless thumb button. Fortunately it as never screwed up the desktop but is definitely useless. It's a very nice feature to be able to use the thumb button in any way with your script !

Is it possible to address the thumb button to any other button function rather then button 5 (I have first to find out which button is which number) ?

Yes, it's shame that a such an worldwide enterprise doesn't support Linux and to be honest I would preferably avoid buying such brands but in my experience these mouses are very long lasting but definitely a shame (even more because I'm living where the brand is from !!!). Money money money...

I don't really need any other features then the basics so, I can live with elementary functions.

Again, thanks a lot.

**Comment by fin-ger** [comment]
> Is it possible to address the thumb button to any other button function rather then button 5 (I have first to find out which button is which number) ?

Yes, you can freely choose between button 1 and 7.

**Comment by sushi2503** [comment]
Perfect, thank you for your work.
